### PR TITLE
Modify delete command

### PR DESCRIPTION
**Pull request by garyljj**
- Add `delete` command without parameters. It clears all person in filtered list

- Rename DeleteTagCommand -> DeleteContactWithTagCommand.
To make way for `delete tag` command.
Did not change functionality (will modify to delete all contact regardless of tags when adding `delete tag`).

- Add deprecated comment to `clear`.
Will remove `clear` later on when tests added for `delete` (ensured that working properly)

part of #102

### CODE REVIEW

**Review by pyuxiang — DISMISSED** [review]
Other than slightly awkward phrasing of "Deleted Person: John, Amy, Coney", otherwise looks fine. Mainly allowing `delete` to accept no arguments, and refactoring `DeleteTagCommand` to `DeleteContactWithTagCommand`.

LGTM.

File: src/main/java/seedu/partyplanet/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -8,10 +8,11 @@
...
-            + ": Deletes the person identified by the index number used in the displayed person list\n"
+            + ": Deletes all contacts in currently filtered list\n"
+            + "OR Deletes the person identified by the index number used in the displayed person list\n"
             + "OR Delete the person with the provided tag,\n"
             + "unless the person has other tags, then only the tag is removed\n"
-            + "Parameters: {INDEX (must be a positive integer) [INDEX]... | -t TAG [-t TAG]...}\n"
+            + "Parameters: [{INDEX (must be a positive integer) [INDEX]... | -t TAG [-t TAG]...}]\n"
             + "Example: " + COMMAND_WORD + " 1 2 3";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
```
Comment (pyuxiang) [inline src/main/java/seedu/partyplanet/logic/commands/DeleteCommand.java:18]: Perhaps a more natural phrasing?

```suggestion
    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted the following people: %1$s";
```

**Review by Ellevy — DISMISSED** [review]
LGTM

File: src/main/java/seedu/partyplanet/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (garyljj) [reply src/main/java/seedu/partyplanet/logic/commands/DeleteCommand.java:18]:
```suggestion
    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted the following persons: %1$s";
```

**Review by Ellevy — APPROVED** [review]
LGTM

**Review by pyuxiang — APPROVED** [review]
LGTM

**Pull request merged**